### ISSUE TITLE
Brand Name

### ISSUE THREAD

**Issue by elliotwaite** [SciNim/flambeau#4]
I'm excited to see this project being developed. I saw that this project was recently rebranded from minitorch to Flambeau. I was wondering what your thoughts were on potentially trying to instead adopt the name NimTorch for this project from the previous effort. Given how popular PyTorch is, if this is going to be the official Nim wrapper for the Torch library, I feel like the name NimTorch would be the most fitting and intuitive, especially for those coming from Python. I'm assuming the possibility of using the NimTorch name would depend on if Giovanni was okay with it, and there'd be some issues to figure out with versioning, but I think this idea would be worth exploring. What do you think?

**Comment by mratsim** [comment]
I'm neither in favour or against. Ping @sinkingsugar

**Comment by elliotwaite** [comment]
Understandable. My preference might just be from seeing the word `torch` so many times in PyTorch code that it feels very familiar to me, and my natural instinct would be to type `import torch` at the top of a file. I'd be interested to hear if others in the Nim science community had a strong preference one way or the other, but in the end, I think your preference is most important since you are the one building this project.

**Comment by sinkingsugar** [comment]
Well, 

First of all the philosophy behind NimTorch was never a wrapper, but rather a Nim implementation on top of the C++ foundation of PyTorch. 
e.g. https://medium.com/@giovanni_94706/almost-full-pytorch-experience-on-ios-by-nimtorch-cadf43b3863b when pytorch didn't even have a C++ library.

Second, the project was a blast and I got to meet the whole PyTorch team in SF and from there even managed to have a few small customers using it. Sadly this did not last long and I had to stop working on it, the Nim community was never really helpful either.

So I'm sorry to say that I don't feel like suddenly all that time and efforts should disappear. I'm currently busy with other things but once nim gets better at managing memory plus some demand NimTorch might still take off again.

Btw I also think `Flambeau` is a very cool name.

**Comment by elliotwaite** [comment]
Makes sense. Thanks for replying @sinkingsugar. And thanks for your efforts with NimTorch. Reading about NimTorch in the PyTorch white paper was the first time I heard about Nim and is what got me interested in the language in the first place. I'm glad you're still interested in potentially continue to work on NimTorch in the future.

**Comment by elliotwaite** [comment]
Flambeau works for me. But I also brainstormed some other potential names that I like as well:
* `libtorch` (precise but less creative)
* Helios (sun god)
* Lux (SI unit for illuminance, but I think you are already using this name inside of laser)
* Lumen (SI unit for luminous flux)

If you want to stick with Flambeau, feel free to close this thread. And thanks for being open to considering an alternative.

**Comment by mratsim** [comment]
We can keep those in mind for the higher level API though I gravitate around Arcanum to keep the Arraymancer theme.

The high-level API would brings at the very least:

- Machine-learning
  - Randomized PCA and fast randomized SVD https://github.com/mratsim/Arraymancer/blob/93200f2/src/arraymancer/ml/dimensionality_reduction/pca.nim#L112-L168
  - Clustering algorithms
  - Missing loss functions (F1-score, top-k, ...)
- Deep-learning
  - What is not in PyTorch or libtorch for example shape inference like in Keras or Arraymancer
- C++
  - We keep raw C++ types like CppVector or CppTuple and gives basic tool to manipulate them from Nim
    but at a higher level we likely want to completely hide them. 

A different library, Agent Smith (https://github.com/numforge/agent-smith) will focus on reinforcement learning. I will likely split the Gym / training environment out of it.

Unsure:
- ggplotnim and Arraymancer compat (not sure where they should live)
- fastText compat https://github.com/Nim-NLP/fastText
- Gradient boosting (xgboost, lightgbm)
- Tensorboard (if it's in PyTorch, maybe directly in Flambeau)
- experimentations, parameters handling (maybe in a dedicated library)

Anyway for those who wants to figure things out, we have a #science room on Discord [![Join the chat on Discord #nim-science](https://img.shields.io/discord/371759389889003530?color=blue&label=nim-science&logo=discord&logoColor=gold&style=flat-square)](https://discord.gg/f5hA9UK3dY) (It's also bridged to Gitter and IRC).
